server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -348,17 +347,8 @@ public void close() throws IOException {
         IOUtils.close(remoteClusters.values());
     }
 
-    public void getRemoteConnectionInfos(ActionListener<Collection<RemoteConnectionInfo>> listener) {
-        final Map<String, RemoteClusterConnection> remoteClusters = this.remoteClusters;
-        if (remoteClusters.isEmpty()) {
-            listener.onResponse(Collections.emptyList());
-        } else {
-            final GroupedActionListener<RemoteConnectionInfo> actionListener = new GroupedActionListener<>(listener,
-                remoteClusters.size(), Collections.emptyList());
-            for (RemoteClusterConnection connection : remoteClusters.values()) {
-                connection.getConnectionInfo(actionListener);
-            }
-        }
+    public Stream<RemoteConnectionInfo> getRemoteConnectionInfos() {
```

Comment:
I'm returning a `Stream` here because I read https://stackoverflow.com/a/24679745/233833 yesterday. The caller always wants a `List` but since we *have* a stream here it seems sane to return the stream and let the caller do what they want with it.

Reply:
None. I wouldn't go out of my way to return a `Stream` if I didn't already have one. I just figured, since I do, I may as well return it.